### PR TITLE
Refactor Repository To Support DID Resolution & Code Generation

### PR DESCRIPTION
**Pull request by LuisOsta**
1. Refactored to Rust monorepo
2. Integrated Buf code generation into Registry Resolver

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,5 @@
...
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
-[dependencies]
-serde_json = "1.0.81"
-signature = "1.5.0"
+[workspace]
+members = [
+	"core",
+	"registry-resolver"
+]
```
Comment (sKudryashov) [inline Cargo.toml:5]: NIT: Newline pls

**Review by sKudryashov — APPROVED** [review]
Fix NIT please

**Review by Ryanmtate — APPROVED** [review]
Looks great-- just a minor suggestion on the use of `#[path]` over `include!`, but can be modified later if you find one to be more useful than the other.

File: registry-resolver/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+pub mod registry {
+    include!("gen/registry_api.v1.rs");
```
[the remark this answers is not in the document]
Comment (LuisOsta) [reply registry-resolver/src/lib.rs]: I didn't run into any issues but I'll migrate it over if that's the better approach

File: registry-resolver/src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,5 @@
-pub mod registry {
-    include!("gen/registry_api.v1.rs");
-}
+#[path = "gen/registry_api.v1.rs"]
+pub mod registry {}
```
Comment (Ryanmtate) [inline registry-resolver/src/lib.rs:2]: okay to drop the `{}`

`pub mod registry` should work by itself.

**Pull request merged**